changelog/fragments/1707852625-Prefer-elastic-agent-client-APMConfig-in-agent-mode.yaml

```diff
@@ -0,0 +1,36 @@
+# Kind can be one of:
+# - breaking-change: a change to previously-documented behavior
+# - deprecation: functionality that is being removed in a later release
+# - bug-fix: fixes a problem in a previous version
+# - enhancement: extends functionality but does not break or fix existing behavior
+# - feature: new functionality
+# - known-issue: problems that we are aware of in a given version
+# - security: impacts on the security of a product or a user’s deployment.
+# - upgrade: important information for someone upgrading from a prior version
+# - other: does not fit into any of the other categories
+kind: enhancement
+
+# Change summary; a 80ish characters long description of the change.
+summary: Prefer elastic-agent-client APMConfig in agent mode
```

Comment:
@michel-laterman Does this diff look good to you? It is strange that a previous pr changes are included

Reply:
I manually removed the commits